project1/server/server_tasks/server_task.cpp

```diff
@@ -0,0 +1,65 @@
+#include "server_task.h"
+namespace server_tasks {
+
+    thread_pool::Task::Status ServerTask::SetUp()  {
+        // Create the socket.
+        const auto kAddress = MakeAddress(port_);
+
+        server_fd_ = SetUpSocket(kAddress);
+        if (server_fd_ < 0) {
+            return thread_pool::Task::Status::FAILED;
+        } else return thread_pool::Task::Status::RUNNING;
```

Comment:
(minor) Let's put brackets around this `else` clause. Or, you can do this with a ternary operator, e.g.
`return server_fd_ < 0 ? thread_pool::Task::Status::FAILED : thread_pool::Task::Status::RUNNING;`